knowageutils/src/main/java/it/eng/spagobi/utilities/rest/RestUtilities.java

```diff
@@ -493,4 +495,44 @@ public static Map<String, String> toHeaders(HttpServletRequest request) {
 		return headers;
 	}
 
+	private static void addHeadersToRequest(Entry<String, String> entry, HttpMethodBase method) {
+		String headerValue = entry.getValue();
+		String headerKey = entry.getKey();
+		headerValue = RestUtilities.sanitizeValueFromRequestHeader(headerValue);
+		RestUtilities.checkIfValueFromRequestHeaderIsInWhitelistFromPropertiesFile(headerKey);
+		method.addRequestHeader(headerKey, headerValue);
+	}
+
+	private static String sanitizeValueFromRequestHeader(String value) {
+		return value.replaceAll("\r", "").replaceAll("%0d", "").replaceAll("%0D", "").replaceAll("\n", "").replaceAll("%0a", "").replaceAll("%0A", "");
+	}
+
+	private static void checkIfValueFromRequestHeaderIsInWhitelistFromPropertiesFile(String value) {
+		List<String> whitelist = RestUtilities.getWhitelistFromPropertiesFile();
+		if (!whitelist.contains(value))
+			throw new SpagoBIRuntimeException("Header value " + value + " is not in the list of allowed headers.");
+	}
+
+	private static List<String> getWhitelistFromPropertiesFile() {
+		Properties properties = new Properties();
+		List<String> whitelist = new ArrayList<>();
+
+		try (InputStream inputStream = RestUtilities.class.getClassLoader().getResourceAsStream("headers.properties")) {
+			if (inputStream != null) {
+				properties.load(inputStream);
+
+				String allowedHeaders = properties.getProperty("allowed.headers");
+				if (allowedHeaders != null && !allowedHeaders.isEmpty()) {
+					whitelist.addAll(Arrays.asList(allowedHeaders.split(",")));
+				}
+			} else {
+				throw new RuntimeException("headers.properties file not found.");
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
We are slowly trying to remove all the printStackTrace() from the code. Please replace it with a runtime Exception like SpagoBiRuntimeException because you are reading a file that need to be present: it's a on-off condition; if it present it's ok, if it's not present we don't have a way to manage the situation. The correct way should be an Error.

Reply:
Fixed